aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+
+- You automatically get the descriptive label value from the ``Display`` attribute. The intended display name might change over time, and the combination of ``Display`` attribute and Label Tag Helper will apply the ``Display`` everywhere it's used.
+- Less markup in source code
+- Strong typing with the model property.
+
+Sample:
+
+.. literalinclude::  forms/sample/final/ViewModels/SimpleViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterLabel.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+
+The following HTML is generated for the ``<label>`` element:
+
+.. code-block:: HTML
+
+ <label for="Email">Email Address</label>  
+ 
+The Label Tag Helper generated the ``for`` attribute value of "Email", which is the ID associated with the ``<input>`` element. The Tag Helpers generate consistent ``id`` and ``for`` elements so they can be correctly associated. The caption in this sample comes from the ``Display`` attribute. If the model didn't contain a ``Display`` attribute, the caption would be the property name of the expression.
+ 
+The Validation Tag Helpers
+---------------------------
+
+There are two Validation Tag Helpers. The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__ (which displays a validation message for a single property on your model), and the `Validation Summary Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationSummaryTagHelper/index.html>`__ (which displays a summary of validation errors). The `Input Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/InputTagHelper/index.html>`__ adds HTML5 client side validation attributes to input elements based on data annotation attributes on your model classes. Validation is also performed on the server. The Validation Tag Helper displays these error messages when a validation error occurs. 
+
+The Validation Message Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Adds the `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__  ``data-valmsg-for="property"`` attribute to the `span <https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span>`__ element, which attaches the validation error messages on the input field of the specified model property. When a client side validation error occurs, `jQuery <https://jquery.com/>`__ displays the error message in the ``<span>`` element. 
+- Validation also takes place on the server. Clients may have JavaScript disabled and some validation can only be done on the server side.
+- HTML Helper alternative: `@Html.ValidationMessageFor <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/HtmlHelperValidationExtensions/index.html>`__ 
```

Comment:
The code displays strangely and looks inconsistent w/ other alternatives in this document: Remove the `@` sign, use double backticks, and don't link the name i.e. make this just
```
... alternative: ``Html.ValidationMessageFor``
```